From c51d8c65ef4bf42524befbf71ea44da09534a83b Mon Sep 17 00:00:00 2001
From: Frediano Ziglio <fziglio@redhat.com>
Date: Thu, 15 Nov 2018 21:55:11 +0000
Subject: [PATCH] udscs: Avoid file descriptor leak

If connection fails the socket descriptor is not closed causing
a leak.

Signed-off-by: Frediano Ziglio <fziglio@redhat.com>
Acked-by: Victor Toso <victortoso@redhat.com>
(cherry picked from commit 2d6d0d2acd3248e981361c2f3d8b253da8f04048)
---
 src/udscs.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/udscs.c b/src/udscs.c
index 288aca2..9901786 100644
--- a/src/udscs.c
+++ b/src/udscs.c
@@ -177,6 +177,7 @@ struct udscs_connection *udscs_connect(const char *socketname,
         if (conn->debug) {
             syslog(LOG_DEBUG, "connect %s: %m", socketname);
         }
+        close(conn->fd);
         free(conn);
         return NULL;
     }
-- 
2.20.1

